### ISSUE TITLE
Segfault appears in Canal pod after Upgrading rancher

### ISSUE THREAD

**Issue by Auston-Ivison-Suse** [rancher/rancher#36181]
**Rancher Server Setup**
- Rancher version:2.5.11 upgraded to 2.5.12-rc4
- Installation option (Docker install/Helm Chart): Docker
- Proxy/Cert Details: self-signed

**Information about the Cluster**
- Kubernetes version: v1.20.14
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): custom using rhel 8.4 ami

   
**Describe the bug**
When upgrading rancher from v2.5.11 to v2.5.12-rc4 we see segfault in the Canal workload under the kube-flannel pod. It appears the segfault has no negative consequences within any deployments. We are opening this issue to make sure we are aware of this in the case any future issues do arise. 

**To Reproduce**
1. Before upgrading rancher deploy custom cluster using 8.4 rhel ami. 
2. Create a deployment in the default namespace. 
3. Check the dmesg in the kube-flannel pod under workload --> canal
4. There will be no segfaults
5. Upgrade rancher and the pods get recreated, which then produce the segfault

**Result**
We see segfaults in the dmesg of the kube flanel pod of the canal workload. 

```
...
[  627.898429] iptables-save[11911]: segfault at 80 ip 00007faf4a33f964 sp 00007ffdbebac838 error 4 in libnftnl.so.11.2.0[7faf4a33a000+19000]
[  627.936197] Code: 83 c4 08 5b 5d 41 5c 41 5d c3 0f 1f 40 00 48 83 c4 08 31 c0 5b 5d 41 5c 41 5d c3 66 66 2e 0f 1f 84 00 00 00 00 00 f3 0f 1e fa <48> 8b 87 80 00 00 00 48 83 ef 80 48 39 f8 74 1b 85 f6 75 0c eb 18

```

**Expected Result**
There shouldn't be any segfaults within the kube flannel pod. 

**Screenshots**

**Additional context**
This will be related to https://github.com/rancher/rancher/issues/36021

**Comment by snasovich** [comment]
Moving to v2.6.6 milestone based on this from the description:
> It appears the segfault has no negative consequences within any deployments. We are opening this issue to make sure we are aware of this in the case any future issues do arise._

**Comment by Raboo** [comment]
I've hit the same issue however, I do not know if this is related or not. But on Flatcar Linux v3185.1.1 or newer networking stops entirely after the iptables starts segfaulting.. So please don't postpone this issue any more.

**Comment by Raboo** [comment]
Is there a way to work around myself this by forcing a newer version of canal or something?

**Comment by snasovich** [comment]
Based on most recent [comment](https://github.com/rancher/rancher/issues/37864#issuecomment-1149949792) to the linked issue segfault is likely somewhat benign. Moving it 2.6.x for now.

**Comment by Josh-Diamond** [comment]
Ticket #36181 - Test Results [pt. 1]

With Docker on a single-node instance:

Reproduction Failure on rancher `v2.5.11` => `v2.5.12-rc4`:
1. Fresh install of rancher `v2.5.11`
2. Provision a HA Custom cluster w/ RHEL 8.4 and k8s `v1.20.15`
3. Deploy a workload in the default namespace
4. Check the `dmesg` logs in the kube-flannel pod under workload --> canal
5. Verified - No segfaults
6. Upgrade rancher from `v2.5.11` => `v2.5.12-rc4`
7. Check the `dmesg` logs in the kube-flannel pod under workload --> canal
8. Reproduction Failure - No segfaults

Screenshots:

`Step 5 - Rancher v2.5.11`
<img width="1628" alt="Screen Shot 2022-06-08 at 11 52 10 AM" src="https://user-images.githubusercontent.com/46494969/172722526-5760ab72-d142-4d56-8a02-7f4419aa48a3.png">

`Step 8 - Rancher v2.5.12-rc4`
<img width="1633" alt="Screen Shot 2022-06-08 at 12 03 12 PM" src="https://user-images.githubusercontent.com/46494969/172722542-215ebd61-0913-4f3d-b843-ca377fb29711.png">

**Comment by Raboo** [comment]
> I've hit the same issue however, I do not know if this is related or not. But on Flatcar Linux v3185.1.1 or newer networking stops entirely after the iptables starts segfaulting.. So please don't postpone this issue any more.

Yes, this was a mistake, network still works, there was another issue that caused me to loose network.

**Comment by Raboo** [comment]
@Josh-Diamond could it be that the segfault happens with newer/older versions of the kernel/iptables? Btw I get this error with Rancher v2.6.3 and v2.6.5

Segfault happens for me on Flatcar Linux with kernel v5.15.43 and v5.15.37. Also on BurmillaOS v2(a fork of RancherOS) with kernel v5.10.28.

**Comment by snasovich** [comment]
Closing this issue as "won't do" as RKE1 is EOL
